javascript/javascript3/week1/GiphyApi/main.js

```diff
@@ -0,0 +1,53 @@
+const apiKey = 'TnGhS9Vu9jrTmAOpBbNXscaRtP8opJNw'
+const giphyName = document.getElementById('giphy-name');
+const btnsearch = document.getElementById('btn-giphy');
+const giphyNum = document.getElementById('giphy-num');
+const displyGiphy = document.getElementById('giphy-result')
+let giphyNameSearch = giphyName.value;
+let disGiphy = giphyNum.value
+giphyNameSearch = "hello";
+disGiphy = 25
+if (giphyNameSearch === "hello") {
+    display()
+}
+function display() {
+    fetch(`https://api.giphy.com/v1/gifs/search?api_key=TnGhS9Vu9jrTmAOpBbNXscaRtP8opJNw&q=${giphyNameSearch}&limit=${disGiphy}&offset=0&rating=g&lang=en`)
+        .then(response => response.json())
+        .then(giphyData => {
+            const data = giphyData.data
+
+            data.forEach(element => {
+                const urlGiphy = element.images.original.url;
+                const imgGiphy = document.createElement('img');
+                imgGiphy.classList.add("img-giphy");
+                imgGiphy.setAttribute("src", urlGiphy);
+                displyGiphy.appendChild(imgGiphy);
+            });
+        })
+}
+btnsearch.addEventListener('click', () => {
+    giphyNameSearch = giphyName.value;
+    if (!giphyNameSearch) {
+        displyGiphy.innerHTML = 'Enter Giphy Name'
+    }
+    else {
+        displyGiphy.innerHTML = ''
+        display();
+        giphyName.value = ''
+    }
+    giphyNum.addEventListener('keyup', () => {
```

Comment:
this eventListener is added with every click. Shouldn't it be outside of the above addEventListener scope

Reply:
I don't understand your point in this comment. I have added inside btnSearch event listener because after search giphy name, user can change number of display Giphy according to requirement. I have having trouble hear, I have used keyup event and I can't write double digit number in input filed.